### PR TITLE
UCT/UCP: Disable interface progress by default.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4751/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4753/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4763/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4764/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4767/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask @brminich @shamisp pls take a look

**Comment by shamisp** [comment]
API change adding @bbenton

**Comment by shamisp** [comment]
@yosefe what is the impact on ugni ?

**Comment by yosefe** [comment]
should no be impact on ugni, but it think would be good if @MattBBaker could give it a try

**Comment by MattBBaker** [comment]
Tested on the Cray machine. Everything works as expected. I'll try running a performance bench next.

**Comment by shamisp** [comment]
👍  @bbenton please review. API change at UCT.

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -2306,11 +2309,15 @@ UCT_INLINE_API ucs_status_t uct_iface_tag_recv_cancel(uct_iface_h iface,
  * Notify the transport that it should actively progress communications during
  * @ref uct_worker_progress().
  *
- * When the interface is created, its progress is enabled.
+ * When the interface is created, its progress is initially disabled.
```
Comment (bbenton) [inline src/uct/api/uct.h]: The corresponding comment for uct_iface_progress_disable needs to be changed to similarly indicate that progress is disabled when the interface is created.

Comment (yosefe) [reply src/uct/api/uct.h]: fixed

**Comment by yosefe** [comment]
@bbenton fixes- can u pls take a look again?

**Comment by bbenton** [comment]
:+1: for Doc/API

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4786/ for details (Mellanox internal link).

**Pull request merged**